lib/product-import.coffee

```diff
@@ -182,6 +186,71 @@ class ProductImport
   _generateUniqueToken: ->
     _.uniqueId "#{new Date().getTime()}"
 
+  _fetchAndResolveCustomReferences: (product) =>
+    new Promise (resolve) =>
+      if product.masterVariant
+        @_fetchAndResolveCustomReferencesByVariant(product.masterVariant).then (result) -> product.masterVariant = result
+
+      if product.variants
+        Promise.map product.variants, (variant) =>
+          @_fetchAndResolveCustomReferencesByVariant(variant)
+          .then ->
+            Promise.resolve()
+        ,{concurrency: 5}
+      resolve(product)
+
+  _fetchAndResolveCustomReferencesByVariant: (variant) =>
+    new Promise (resolve) =>
+      if variant.attributes and not _.isEmpty(variant.attributes)
+        _.map variant.attributes, (attribute) =>
+          if attribute and _.isArray(attribute.value)
+            if _.every(attribute.value, @_isReferenceTypeAttribute) # all elements in the attribute array should be a ref.
+              @_resolveCustomReferenceSet(attribute.value)
+              .then (result) ->
+                attribute.value = result
+                resolve(variant)
```

Comment:
This is also wrong, `resolve` should be called only once. The all promise handling is wrong here